tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java

```diff
@@ -120,7 +121,10 @@ public <T> T getJaxbRoot(InputStream stream, Class<T> clazz, Schema schema)
 		ContextProvidingValidationEventHandler handler = new ContextProvidingValidationEventHandler();
 		try {
 			staxEventReader = new JpaNamespaceTransformingEventReader( staxEventReader );
-			JAXBContext jaxbContext = JAXBContext.newInstance( ObjectFactory.class );
+			ClassLoader cl = this.getClass().getClassLoader();
+			String packageName = Optional.ofNullable(ObjectFactory.class.getPackage())
+					.map(p -> p.getName()).orElse("org.hibernate.processor.xml.jaxb");
```

Comment:
```suggestion
String packageName  = "org.hibernate.processor.xml.jaxb";
```

Seems like it should do the exact same thing.